watertap/tools/oli/watertap_to_oli_helper_functions.py

```diff
@@ -0,0 +1,195 @@
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+#################################################################################
+
+"""
+This file contains methods to convert WaterTAP naming conventions to OLI
+and generate molecular weight and charge dictionaries from molecular formulae. 
+
+It calculates molecular weights using the periodic_table.csv from:
+https://gist.github.com/GoodmanSciences/c2dd862cd38f21b0ad36b8f96b4bf1ee.
+"""
+
+__author__ = "Paul Vecchiarelli"
+
+from collections import namedtuple
+from pandas import read_csv
+from re import findall
+
+OLIName = namedtuple(
+    "OLIName", ["oli_name", "watertap_name", "charge", "charge_group", "molar_mass"]
+)
+
+
+def watertap_to_oli(watertap_name: str) -> OLIName:
+    """
+    This method creates a named tuple
+    which can be passed directly into OLI
+    or into MCAS property models.
+
+    Parameters
+    ----------
+    watertap_name : str
+        a substance name in WaterTAP format (typical IDAES convention, i.e., B[OH]4_-)
+
+    Raises
+    ------
+    RuntimeError: this generic error is raised if:
+        - watertap_name is a blank string
+        - an extracted charge_sign is not plus or minus
+        - if more than 1 "_" is present in watertap_name
+
+    Returns
+    -------
+    OLIName: named tuple containing:
+        - oli_name: substance name converted to OLI convention
+        - watertap_name: original name passed into method
+        - charge: charge value
+        - charge_group: categorization under one of: Cations, Anions, Neutrals
+        - molar_mass: weight of 1 mole of the substance (grams per mole)
+
+    """
+    components = watertap_name.split("_")
+
+    # neutral molecule
+    if len(components) == 1:
+        if components[0] != "":
+            molecule = components[0]
+            charge = 0
+        else:
+            raise RuntimeError("Unrecognized component " + watertap_name)
+
+    # charged molecule
+    elif len(components) == 2:
+        molecule = components[0] + "ION"
+
+        charge = components[1]
+        charge_sign = charge[-1]
+
+        if len(charge) > 1:
+            charge_magnitude = int(charge[:-1])
+        else:
+            charge_magnitude = 1
+
+        if charge_sign == "+":
+            charge = charge_magnitude
+        elif charge_sign == "-":
+            charge = -charge_magnitude
+        else:
+            raise RuntimeError("Unrecognized charge magnitude " + charge_magnitude)
+
+    else:
+        raise RuntimeError("Unrecognized name format " + watertap_name)
```

Comment:
I think this can be broken off into its own function so that we have helper functions explicitly for grab mw data and determining charge--though the latter is more strongly hinged to naming convention and we'd have to clearly document that discrepancy. On a side note, I think the general guidance for users should be--you don't _have_ to follow any particular naming conventions for solutes if you don't want to, but we have these helper functions the can make life easier (hopefully), and if you want use them, then you do have to follow a particular naming convention.

Overall, the more we can decrease dependence on precise naming, the better.

Reply:
This method currently requires WaterTAP names to compute charge and molar mass, but I think conversion to OLI names could be made optional